### PR TITLE
Upgrade Istio to 1.9.1

### LINKED ISSUE #10865: Upgrade Istio to 1.9.1
**Description**

Upgrade Istio to 1.9.1

AC:

- istio running on 1.9.1 at least
- updated istiooperator resources
- security scans green
- release notes
- all sidecars runs on 1.9.1 (include improvements in proxy-reset-job)

**Reasons**

For security & support reasons we should be up to date with the latest Istio release.

Attachments

https://istio.io/latest/news/releases/1.9.x/announcing-1.9/change-notes/
https://istio.io/latest/news/releases/1.9.x/announcing-1.9/upgrade-notes/

### PR DESCRIPTION
**Pull request by PhillipAmend**
**Description**

Changes proposed in this pull request:

- Upgrade Istio from verstion 1.8.2 to 1.9.1
- Remove depricated 'meshExpansion' setting in the Istio Operator files
- Remove istio-coredns plugin because it's deprecated and not longer necessary. DNS is built into the istio agent in the sidecar now.
- 
**Related issue(s)**
Resolves #10865

### CODE REVIEW

File: resources/istio/files/istio-operator-cluster-evaluation.yaml [hunk 1]
```diff
@@ -150,11 +148,6 @@ spec:
           timeoutSeconds: 5
         hpaSpec:
           maxReplicas: 1
-          metrics:
```
Comment (strekm) [inline resources/istio/files/istio-operator-cluster-evaluation.yaml:153]: to check if syntax changed and what are the default values

File: resources/istio/files/istio-operator-cluster-evaluation.yaml [hunk 2]
```diff
@@ -178,7 +171,10 @@ spec:
     accessLogFile: ""
     trustDomain: cluster.local
     defaultConfig:
-      proxyMetadata: {}
+      proxyMetadata:
```
Comment (strekm) [inline resources/istio/files/istio-operator-cluster-evaluation.yaml:174]: in default manifest this is not present, please check if this setting needs to be set specifically

Comment (PhillipAmend) [reply resources/istio/files/istio-operator-cluster-evaluation.yaml:174]: This is the setting for the coredns addin replacement. It is necessary.

**Review by mjakobczyk — APPROVED** [review]

**Review by mjakobczyk — APPROVED** [review]

**Pull request merged**